lib/db/schemaupdater.go

```diff
@@ -20,21 +18,14 @@ import (
 //   0: v0.14.0
 //   1: v0.14.46
 //   2: v0.14.48
-//   3: v0.14.49
-//   4: v0.14.49
-//   5: v0.14.49
+//   3-5: v0.14.49
 //   6: v0.14.50
 //   7: v0.14.53
-//   8: v1.4.0
-//   9: v1.4.0
+//   8-9: v1.4.0
+//   10: v1.4.1
 const (
-	dbVersion             = 9
-	dbMinSyncthingVersion = "v1.4.0"
-)
-
-var (
-	errFolderIdxMissing = errors.New("folder db index missing")
-	errDeviceIdxMissing = errors.New("device db index missing")
+	dbVersion             = 10
+	dbMinSyncthingVersion = "v1.4.1"
```

Comment:
Acksherly no need to bump the min version

Reply:
Eh, looking closer on this I realized db schema versions don't actually work like I thought they worked. Indeed with this bump to 10 it won't be possible to open the database with 1.4.0 any more, even though the on disk format is in fact the same. We should have something like a compatibility version different from the db version, or actually compare the min db version to our running version to determine if we're ok to proceed... Possibly not for this PR though.

Reply:
Thinking one step further I think this should maybe not be a "migration" at all as it doesn't migrate, rather it should be run with the metadata refresh (periodically and when crash is detected) and triggered as it is from the index sender?

Reply:
Missed that comment:  
I wouldn't do it periodically, as it is rather heavy (iterates over all local file and sequence entries and unmarshalling every file on the way), but only triggered by crash and index sender. I'd still keep the migration, even though it's a misnomer, as otherwise users will get the warning in the UI for a problem that we know has existed and is mitigated with the migration.

Reply:
If it were just a misnomer I'd be fine with it, but it forces a compatibility change and prevents 1.4.0 from opening the db. I think the GUI issue is similar to out of date metadata which we handle by the periodic metadata recalculation which is equally heavy? We could bump the key that stores that recalculation date, causing a recalc on upgrade without a migration.

Reply:
Ah right, the `dbMinSyncthingVersion` implementation doesn't actually do what it should as you wrote above. That would however be an easy fix, so I might just do that in this PR?

It is pretty much exactly the same as the metadata thing (well twice as heavy due to double pass). I agree it's doable, I am just hesitant to add periodic mitigation for something that isn't needed as far as we know today (there are no known sequence error modes and the index sender should pick up any errors which we aren't aware of) - even if the performance impact is reasonable.

Reply:
Well there are no known failure modes to get the metadata out of sync either, yet we do that periodically, and similarly we transmit the whole index on upgrade for some historical reason... I think we should run this check when recovering from a crash, and then either on upgrade (every upgrade) or periodically (STRECHECKDBEVERY). Potentially the "recheck db every ..." should go away and just run on upgrades that too.

Reply:
I'd rather not complicate the migration check, especially not in a 1.4.1 release. It's easier to make the non-migration be a non-migration :)

Reply:
done - periodically, because it keeps historic unknown weirdness together xD